FreeRTOS/FreeRTOS
# Build: Upgrade GCC project files for e2 studio v7.8.0 in Demo/RX600_RX64M_RSK_GCC_e2studio folder

**This PR adds new change to .project of the following PR therefore the following PR has to be merged before this PR.**

Build: Fix Demo/RX600_RX64M_RSK_GCC_e2studio RTOS demo project build error #245

-----------
The following Description, Test Steps, Related Issue and Additional Information are the same as 'Build: Upgrade GCC project files for e2 studio v7.8.0 in Demo/RX200_RX231-RSK_GCC_e2studio_IAR folder #255' other than project folder name, zip file name, some screen copies of e2 studio project settings and the following things.

Original e2 studio version:
```
Demo/RX200_RX231-RSK_GCC_e2studio_IAR       ==> e2 studio v4.0.2.008

Demo/RX600_RX64M_RSK_GCC_e2studio (this PR) ==> e2 studio v3.1.2.09
```
Original GNURX version:
```
Demo/RX200_RX231-RSK_GCC_e2studio_IAR       ==> KPIT GNURX v15.01-SP1 (or v15.01)

Demo/RX600_RX64M_RSK_GCC_e2studio (this PR) ==> KPIT GNURX v14.01
```
Fewer removed file in this PR:
```
Demo/RX200_RX231-RSK_GCC_e2studio_IAR       ==> .settings/Dependency_Scan_Preferences.prefs is removed.

Demo/RX600_RX64M_RSK_GCC_e2studio (this PR) ==> .settings/Dependency_Scan_Preferences.prefs doesn't exist.
```
Description
-----------
Renesas e2 studio v7.8.0 (to be precise, v6.0 or later) requests us to upgrade project which was created by e2 studio v5.4 or earlier. Nevertheless, in case of GCC project, e2 studio doesn't upgrade the project correctly, so that modifying the upgraded project by hand is mandatory to build the project successfully.

This PR upgrades and fixes such GCC project in Demo/RX600_RX64M_RSK_GCC_e2studio and makes the project able to be built successfully using e2 studio v7.8.0 (to be precise, v6.0 or later).

But, after this PR is applied, the project is unable to be built using e2 studio v5.4 or earlier. If this isn't acceptable for you, please let me know.

**This PR updates the following files:**

.project
.cproject
.settings/language.settings.xml
RTOSDemo HardwareDebug.launch

**This PR adds the following files:**

src/linker_scriptHardwareDebug.ld

**This PR removes the following files which are no longer necessary (no longer generated when creating new project):**

.HardwareDebuglinker
.info
custom.bat
makefile.init
.settings/Project_Generation_Preferences.prefs

Test Steps
-----------
At first, I built this project using the following source code and tools to get a MOT file.

COMBINATION 1)

(S) https://github.com/FreeRTOS/FreeRTOS-Kernel/commit/82fdc1c3ee4c5e8a1f10e2a2370c81088506e905 (Aug 29, 2020), FreeRTOS/FreeRTOS https://github.com/FreeRTOS/FreeRTOS/commit/bcd5dec6c46392c8ab30a091dda65379d46a2fb5 (Aug 29, 2020)
(O) Without this PR
(1) e2 studio v3.1.2.09 and KPIT GNURX v14.01 and KPIT OPTLIB

Then, I built this project using the following source code and tools to get new MOT file and I confirmed that there are no changes in new MOT file.

COMBINATION 2)

(S) same source code
(P) With this PR
(2) e2 studio v7.8.0 and KPIT GNURX v14.01 and KPIT OPTLIB

COMBINATION 3)

(S) same source code
(P) With this PR
(3) e2 studio v2020-07 (64bit version) and KPIT GNURX v14.01 and KPIT OPTLIB

And I built this project using the following source code and tools and I confirmed that this project is able to be built.

COMBINATION 4)

(S) same source code
(P) With this PR
(P') Additionally, toolchain type and toolchain version are re-selected.
(4) e2 studio v7.8.0 and GCC for Renesas RX 8.3.0.202002 and KPIT OPTLIB

COMBINATION 5)

(S) same source code
(P) With this PR
(P') Additionally, toolchain type and toolchain version are re-selected.
(5) e2 studio v2020-07 (64bit version) and GCC for Renesas RX 8.3.0.202002 and KPIT OPTLIB

But I don't have the evaluation board therefore I can't check the program which are built.
If this isn't acceptable for you, please let me know.

Related Issue
-----------
A report of build check result using FreeRTOS V10.3.1 and CC-RX/GNURX/ICCRX for Renesas RX MCUs #154
https://github.com/FreeRTOS/FreeRTOS/issues/154#issuecomment-687141904

By submitting this pull request, I confirm that you can use, modify, copy, and redistribute this contribution, under the terms of your choice.

Additional Information
-----------
The following ZIPs contain MOT, MAP and build.log files which were built using COMBINATION 1) and COMBINATION 2).

[RX600_RX64M_RSK_GCC_e2studio_Demo_MOT_e2_3_1_2_09_GNURX_14_01_optlib__20200906.zip](https://github.com/FreeRTOS/FreeRTOS/files/5209058/RX600_RX64M_RSK_GCC_e2studio_Demo_MOT_e2_3_1_2_09_GNURX_14_01_optlib__20200906.zip)
[RX600_RX64M_RSK_GCC_e2studio_Demo_MOT_e2_7_8_0_GNURX_14_01_optlib__20200909.zip](https://github.com/FreeRTOS/FreeRTOS/files/5209062/RX600_RX64M_RSK_GCC_e2studio_Demo_MOT_e2_7_8_0_GNURX_14_01_optlib__20200909.zip)

To build the same MOT file, the following unusual settings are used.

![e2studio_freertos_build_issues_131](https://user-images.githubusercontent.com/22914520/92927619-e432aa80-f478-11ea-8900-c83ba50f2711.png)
![e2studio_freertos_build_issues_132](https://user-images.githubusercontent.com/22914520/92927635-ea288b80-f478-11ea-802b-bb89c30dcab6.png)

Toolchain type and toolchain version are as follows.

COMBINATION 2) and COMBINATION 3) and also COMBINATION 1)
![e2studio_freertos_build_issues_133](https://user-images.githubusercontent.com/22914520/92927644-ed237c00-f478-11ea-8113-b36fb0faed3b.png)

COMBINATION 4) and COMBINATION 5)
![e2studio_freertos_build_issues_104](https://user-images.githubusercontent.com/22914520/92721715-7798ac80-f3a1-11ea-9625-0900d40879a4.png)
